src/Log/Engine/BaseLog.php

```diff
@@ -61,6 +68,23 @@ public function __construct(array $config = [])
         if (!empty($this->_config['types']) && empty($this->_config['levels'])) {
             $this->_config['levels'] = (array)$this->_config['types'];
         }
+
+        // Default to DefaultFormatter for BC with custom loggers that don't set `formatter` in $_defaultConfig
+        $formatterConfig = $this->_config['formatter'] ?? DefaultFormatter::class;
```

Comment:
Should we also allow the formatter to be passed as an instance? We do that for log/cache engines and it helps when someone wants to handle construction via a DI container or some more elaborate solution.